cdap-event-reader-spi/src/main/java/io/cdap/cdap/spi/events/EventReader.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.spi.events;
+
+
+/**
+ * {@link EventReader} Interface for listening for events.
+ */
+public interface EventReader<T extends Event> extends AutoCloseable {
+  /**
+   * Method to initialize EventReader.
+   *
+   * @param context Configuration properties of reader
+   */
+  void initialize(EventReaderContext context);
+
+  /**
+   * Pull messages if available.
+   *
+   * @param maxMessages maximum messages to pull
+   * @return Result containing events
+   */
+  EventResult<T> pull(int maxMessages);
+
+  /**
+   * Returns the identifier for this reader.
+   *
+   * @return String id for the reader
+   */
+  String getId();
+
+  /**
+   * Close connection.
```

Comment:
This javadoc seems tied to implementation as not all implementations has "connection". For interface javadoc, it is important to describe the contract. For example in this case, it is better as "Invoked as the last method call to this instance. Implementation can leverage this call for freeing up underlying resources".